pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -26,17 +26,18 @@ type TemplateType string
 
 // Possible template types
 const (
-	TemplateTypeContainer    TemplateType = "Container"
-	TemplateTypeContainerSet TemplateType = "ContainerSet"
-	TemplateTypeSteps        TemplateType = "Steps"
-	TemplateTypeScript       TemplateType = "Script"
-	TemplateTypeResource     TemplateType = "Resource"
-	TemplateTypeDAG          TemplateType = "DAG"
-	TemplateTypeSuspend      TemplateType = "Suspend"
-	TemplateTypeData         TemplateType = "Data"
-	TemplateTypeHTTP         TemplateType = "HTTP"
-	TemplateTypePlugin       TemplateType = "Plugin"
-	TemplateTypeUnknown      TemplateType = "Unknown"
+	TemplateTypeContainer            TemplateType = "Container"
+	TemplateTypeContainerSet         TemplateType = "ContainerSet"
+	TemplateTypeSteps                TemplateType = "Steps"
+	TemplateTypeScript               TemplateType = "Script"
+	TemplateTypeResource             TemplateType = "Resource"
+	TemplateTypeMulticlusterResource TemplateType = "MulticlusterResource"
```

Comment:
I’m not sure about this “multi-cluster resource”. I’m not sure we need it. Can you do in another PR to discus?

Reply:
For sure. Let's sync about it. I'll put some time on your calendar